### ISSUE TITLE
SimpleXMLElement regressions on Psalm 5.15

### ISSUE THREAD

**Issue by edsrzf** [vimeo/psalm#10204]
I'm running into a couple regressions related to SimpleXMLElement when trying to upgrade to Psalm 5.15:

* False positive `RedundantCondition` errors when calling `isset` on property fetches
* Less specific return type for the `asXml` method when no arguments are passed

Example code:

```php
<?php

function f(): string|false
{
    $x = new SimpleXMLElement('<a></a>');
    // False positive: RedundantCondition
    if (isset($x->foo)) {
        echo 'hi';
    }
    // Inferred type not specific enough
    return $x->asXml();
}
```

For some reason I can't reproduce either issue on psalm.dev, but I've created a repository here: https://github.com/edsrzf/psalm-xml-repro

I'm guessing these are both due to #10049.

**Comment by edsrzf** [comment]
I think the problem is that I didn't have the `simplexml` extension enabled, by either adding `ext-simplexml` as a dependency in `composer.json` or via `enableExtensions` in `psalm.xml`. With either of those, the errors go away.

I guess this makes sense but I find it surprising to have to make this change in a minor version update.

**Comment by robchett** [comment]
I'm not 100% sure, but I think the online tool has all extensions enabled. At the very least it appears to have simplexml. Which is why this can't be reproduced there.

Before a4de6d9dc092d77871b6097af42498a192362e9a `SimpleXMLElement` & `SimpleXMLIterator` were defined as universal_object_crates, even if the extension was not explicitly enabled, and reflection was used for the signatures.

I don't think it's right to reintroduce the definitions without explicit loading of the extension, so would class this as a bugfix instead of a regression.

**Comment by edsrzf** [comment]
I agree there's probably nothing to fix here. The BC break is mentioned in the linked PR and hopefully anyone running into it will find this issue.

It might be nice to update the 5.15.0 release notes to mention this more clearly though.
